controllers/slicegateway/utils.go

```diff
@@ -25,19 +25,31 @@ import (
 	"github.com/kubeslice/worker-operator/pkg/hub/controllers"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 func (r *SliceGwReconciler) cleanupSliceGwResources(ctx context.Context, slicegw *kubeslicev1beta1.SliceGateway) error {
 	//delete gateway secret
-	meshSliceGwCerts := &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      slicegw.Name,
-			Namespace: controllers.ControlPlaneNamespace,
-		},
+	listOpts := []client.ListOption{
+		client.InNamespace(slicegw.Namespace),
+		client.MatchingLabels{"kubeslice.io/slice-gw": slicegw.Name},
 	}
-	if err := r.Delete(ctx, meshSliceGwCerts); err != nil {
-		r.Log.Error(err, "Error Deleting Gateway Secret while cleaning up.. Please Delete it before installing slice again")
-		return err
+	secretList := &corev1.SecretList{}
+	if err := r.List(ctx, secretList, listOpts...); err != nil {
+		r.Log.Error(err, "Failed to list gateway secrets")
+
+	}
+	for _, v := range secretList.Items {
+		meshSliceGwCerts := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      v.Name,
+				Namespace: controllers.ControlPlaneNamespace,
+			},
+		}
+		if err := r.Delete(ctx, meshSliceGwCerts); err != nil {
+			r.Log.Error(err, "Error Deleting Gateway Secret while cleaning up.. Please Delete it before installing slice again")
```

Comment:
I think an event would be nice to have here!

Reply:
added